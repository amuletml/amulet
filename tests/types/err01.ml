external val ( + ) : int -> int -> int = ""

type queue 'a
type option 'a = Some of 'a
let rec enqueue : forall 'a 'b. ('a -> option ('a * 'b)) -> 'a -> queue 'b =
  fun f x -> enqueue f x

let range start stop =
  let go (c, lim) = Some (c + 1, lim + 0)
  enqueue go (start, stop)

(* Previously, this would error with
 *
 * q.ml[58:11 ..58:12]: error
 *   Could not match expected type option ('a * 'b) with option (int * int)
 *
 *   Arising in the expression
 *    │
 * 58 │   enqueue go (start, stop)
 *    │
 * or something like, at least. Regression test to make sure we don't break that
 * again.
 *)
