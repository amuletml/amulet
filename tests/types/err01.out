err01.ml[8:11 ..8:12]: error
  Could not match expected type option ((int * int) * 'b) with option (int * int)

  Arising in the expression
  │ 
8 │   enqueue go (start, stop)
  │           ^^
